### PR TITLE
change JobPrio to int value

### PR DESCRIPTION
**Pull request by ticoann**
from the report, https://cms-logbook.cern.ch/elog/Workflow+processing/19692

### CODE REVIEW

**Comment by ticoann** [comment]
@bbockelm, Hi Brian, please review.

**Comment by amaltaro** [comment]
We could cast again priority to integer, but it shouldn't be needed.
Pushing this patch to the production nodes since JobUpdater is winning a lot.
What surprises me is that we did not hit this problem in testbed agents (which are also connected to the global pool...)

File: src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py [hunk 1]
```diff
@@ -858,7 +858,7 @@ def updateJobInformation(self, workflow, task, **kwargs):
             priority = (int(kwargs['requestPriority']) + int(kwargs['taskPriority'])*self.maxTaskPriority)
             try:
                 sd.edit('WMAgent_JobID =!= "UNDEFINED" && WMAgent_SubTaskName == %s && WMAgent_RequestName == %s'% (classad.quote(str(task)),classad.quote(str(workflow))),
-                        "JobPrio", classad.ExprTree('"%s"'% priority))
+                        "JobPrio", classad.ExprTree('"%d"'% priority))
```
Comment (bbockelm) [inline src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py]: I think this is wrong.  I think it should be along the lines of:

classad.Literal(int(priority))

The current patch casts the priority to a string.

**Comment by ticoann** [comment]
Thanks Brian, I updated the patch. Could you point me the condor python wrapper api document? I have no idea, I guess it is time for me to learn. :smile:

**Comment by bbockelm** [comment]
Hi,

The python wrapper API docs are here:

http://research.cs.wisc.edu/htcondor/manual/v8.3/6_7Python_Bindings.html

They are in the HTCondor manual format (which is a bit strange way to do python docs, but it is what we have...) and are reasonably complete.  The online python help function is probably the most accurate.

Brian

**Comment by ticoann** [comment]
Thanks Brian

**Comment by ticoann** [comment]
Actually the problem seems to be this line.
https://github.com/dmwm/WMCore/blob/1.0.1/src/python/WMCore/BossAir/Plugins/CondorPlugin.py#L801

``` python
priority = (int(kwargs['requestPriority']) + int(kwargs['taskPriority']) * self.maxTaskPriority)
```

default self.maxTaskPriority is 1e7 which is float and give the priority to float.
And it seems we were using default value as well before. (There is not maxTaskPriority is set)
https://github.com/dmwm/WMCore/blob/1.0.1/etc/WMAgentConfig.py

However, that code hasn't changed between versions either.  I am not sure why it only causing the problem now.

**Comment by ticoann** [comment]
Alan, do you want me to create new patch for 1.0.5 version?

**Comment by amaltaro** [comment]
I think it would be more clear to have this patch in that branch as well.
OR, we could accumulate a few PR and then cherry pick all these changes for a patch2 for example.

**Pull request merged**